usecases/scheduled_execution/batch_filtering.go

```diff
@@ -0,0 +1,185 @@
+package scheduled_execution
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
+	"github.com/checkmarble/marble-backend/pure_utils/duration"
+
+	"github.com/jackc/pgx/v5"
+)
+
+func selectFiltersFromTriggerAstRootAnd(node ast.Node, table models.TableIdentifier) []models.Filter {
+	if node.Function != ast.FUNC_AND {
+		return nil
+	}
+
+	filters := make([]models.Filter, 0, 10)
+	for _, node := range node.Children {
+		filter, valid := astConditionToFilter(node, table)
+		if valid {
+			filters = append(filters, filter)
+		}
+	}
+	return filters
+}
+
+func astConditionToFilter(node ast.Node, table models.TableIdentifier) (models.Filter, bool) {
+	switch node.Function {
+	case ast.FUNC_GREATER,
+		ast.FUNC_GREATER_OR_EQUAL,
+		ast.FUNC_LESS,
+		ast.FUNC_LESS_OR_EQUAL,
+		ast.FUNC_EQUAL,
+		ast.FUNC_NOT_EQUAL:
+		return filterFromComparisonNode(node, table)
+	case ast.FUNC_IS_IN_LIST,
+		ast.FUNC_IS_NOT_IN_LIST,
+		ast.FUNC_CONTAINS_ANY,
+		ast.FUNC_CONTAINS_NONE:
+		return models.Filter{}, false
+	case ast.FUNC_STRING_CONTAINS,
+		ast.FUNC_STRING_NOT_CONTAIN:
+		return models.Filter{}, false
+	}
+	return models.Filter{}, false
+}
+
+func filterFromComparisonNode(node ast.Node, table models.TableIdentifier) (models.Filter, bool) {
+	if !slices.Contains([]ast.Function{
+		ast.FUNC_GREATER,
+		ast.FUNC_GREATER_OR_EQUAL,
+		ast.FUNC_LESS,
+		ast.FUNC_LESS_OR_EQUAL,
+		ast.FUNC_EQUAL,
+		ast.FUNC_NOT_EQUAL,
+	}, node.Function) {
+		return models.Filter{}, false
+	}
+
+	leftVal := comparisonValueFromNode(node.Children[0], table)
+	if !leftVal.valid {
+		return models.Filter{}, false
+	}
+
+	rightVal := comparisonValueFromNode(node.Children[1], table)
+	if !rightVal.valid {
+		return models.Filter{}, false
+	}
+
+	if !leftVal.involvesPayload && !rightVal.involvesPayload {
+		return models.Filter{}, false
+	}
+
+	return models.Filter{
+		LeftSql:    leftVal.rawSql,
+		LeftValue:  leftVal.value,
+		Operator:   comparisonAstNodeToString(node),
+		RightSql:   rightVal.rawSql,
+		RightValue: rightVal.value,
+	}, true
+}
+
+func comparisonAstNodeToString(node ast.Node) string {
+	switch node.Function {
+	case ast.FUNC_GREATER:
+		return ">"
+	case ast.FUNC_GREATER_OR_EQUAL:
+		return ">="
+	case ast.FUNC_LESS:
+		return "<"
+	case ast.FUNC_LESS_OR_EQUAL:
+		return "<="
+	case ast.FUNC_EQUAL:
+		return "="
+	case ast.FUNC_NOT_EQUAL:
+		return "!="
+	default:
+		return ""
+	}
+}
+
+type parsedFilter struct {
+	rawSql          string
+	value           any
+	involvesPayload bool
+	valid           bool
+}
+
+func payloadFieldNodeToSql(node ast.Node, table models.TableIdentifier) (string, bool) {
+	field, ok := node.Children[0].Constant.(string)
+	return pgx.Identifier.Sanitize([]string{table.Schema, table.Table, field}), ok
+}
+
+func comparisonValueFromNode(node ast.Node, table models.TableIdentifier) parsedFilter {
```

Comment:
This function is really complicate, and i think I do not understand the need for `involvesPayload`

Reply:
The idea behind `involvesPayload` is: I don't care about any "lines" in the list of AND conditions that don't involve the payload, as by definition they won't lead to any filters on the table being read.
Think of things like parent (linked) fields (since we don't do a join for the filter), aggregates, or such.

Reply:
I'll see when I'm done if I see a good refactoring of the function - though TBH right now it's already split into 4 well separated cases

Reply:
About `involvesPayload`, I guess in hindsight it's perhaps not strictly necessary if I already reject any condition that uses an aggregation or "linked field" lookup. That would just mean ignoring vs not ignoring conditions that are "constant" - like constant to constant comparison. I suppose I could just remove the condition on `involvesPayload`.

Reply:
I removed `involvesPayload`